File: runtime/java/titan-ast-runtime/src/titan/ast/runtime/AstToken.java
package titan.ast.runtime;

/**
 * 识别文本（通常为源代码）的token.
 *
 * @author tian wei jun
 */
public class AstToken {
  public int start = 0;
  public String text = "";

  public AstToken(int start) {
    this.start = start;
  }

  public AstToken(int start, String text) {
    this.start = start;
    this.text = text;
  }

  @Override
  public String toString() {
    return text;
  }
}

File: runtime/cpp/titan-ast-runtime-lib/AstToken.cpp
//
// Created by tian wei jun on 2023/6/26.
//

#include "AstToken.h"

AstToken::AstToken() : start(0), text(std::string()) {}

AstToken::AstToken(int start, std::string text) : start(start), text(text) {}

AstToken::AstToken(const AstToken &token) = default;

AstToken::AstToken(AstToken &&token) noexcept = default;

AstToken &AstToken::operator=(const AstToken &other) = default;

AstToken::~AstToken() = default;

const AstToken *AstToken::clone() const {
  auto *token = new AstToken(this->start, this->text);
  return token;
}
